### PR TITLE
feature: Add support for upserting to an empty Glue table

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Feature

### Detail
The `s3.merge_upsert_table` function currently does not work when the destination Glue table is empty. If the Glue table is already created, we should be able to simply insert the data.

### Relates
Closing #1562 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 6072b3202a17e4850e3d9d2be5368d288227bb26
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f2171e73-5e2d-40b9-ab7d-6353a79fbe71%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/s3/_merge_upsert_table.py [hunk 1]
```diff
@@ -109,10 +121,18 @@ def merge_upsert_table(
...
-        # Read the existing table into a pandas dataframe
-        existing_df = wr.s3.read_parquet_table(database=database, table=table, boto3_session=boto3_session)
+
+        try:
+            # Read the existing table into a pandas dataframe
+            existing_df = wr.s3.read_parquet_table(database=database, table=table, boto3_session=boto3_session)
+
+        except NoFilesFound:
+            # Generate empty frame with the schema specified in the Glue catalog
+            existing_df = _generate_empty_frame_for_table(database=database, table=table, boto3_session=boto3_session)
```
Comment (LeonLuttenberger) [inline awswrangler/s3/_merge_upsert_table.py:131]: The reason I went for generating an empty DataFrame is to be able to take advantage of the data quality tests, including the schema checks. If the table definition in the Glue catalog is incompatible with `delta_df`, the function should throw an error.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b23b4603884a676108badc65e372f8ca8da373a5
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c6397007-8a43-40ae-928a-0e4e8fe4165d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by borodark** [comment]
Appretiated the reaction time!

**Pull request merged**